src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -491,6 +493,8 @@ private static void reconcileRunsAndBuilds() {
       BuildList buildList = getAuthenticatedOpenShiftClient().builds()
         .inNamespace(property.getNamespace()).withLabel("buildconfig=" + property.getName()).list();
 
+      BuildConfig buildConfig = property.getBuildConfig();
```

Comment:
in line with @akram 's comment below, you should be getting a fresh version of the buildconfig vs. pulling the copy stored in the property object